indicator-stickynotes.py

```diff
@@ -323,12 +327,47 @@ def is_running():
 if __name__ == "__main__":
     import sys
 
-    # if already running send signal 
-    # to make stickynotes appears on top
-    # and exit
-    pid = is_running()
-    if pid:
-        os.kill(pid,signal.SIGUSR1)
-        sys.exit(0)
+    parser = argparse.ArgumentParser(description=_("Sticky Notes"),
+            usage = "%(prog)s [-k | -r | [[-c CATEGORY] " +
+                    "(-i INPUTFILE | -n STRING ...)] | -d]")
+
+    parser.add_argument("-k","--kill", action="store_true",
+            help="kill background proccess")
+    parser.add_argument("-r","--refresh", action="store_true",
+            help="refresh data")
+    parser.add_argument("-c","--category", nargs=1, default=[''],
+            help="using with [-n|-i ...], set categeory", type=str)
+    parser.add_argument("-i","--infile", type=argparse.FileType('r'),
+            help="new sticky note with content from a file")
+    parser.add_argument("-n","--new", metavar='NEW_NOTE', nargs='+',
+            help="create a new note")
+    parser.add_argument("--no-daemon", action="store_true",
+            help="do not daemonize")
+    parser.add_argument("-d", action='store_true',
+            help="use the development data file")
+    args = parser.parse_args()
+
+    if args.new or args.infile: # create a new note if required
+        args.refresh=True
+        noteset=NoteSet(indicator=None, gui_class=None,
+            data_file=stickynotes.info.DEBUG_SETTINGS_FILE if args.d        \
+                else stickynotes.info.SETTINGS_FILE)
+        try: noteset.open()
+        except Exception as e:
+            print('fialed to load config file')
+            sys.exit(1)
+        notebody = ' '.join(args.new).encode().decode('unicode-escape')     \
+            .encode('latin1').decode('utf-8') if args.new                   \
+            else args.infile.read().rstrip()
+        noteset.new(notebody=notebody, category=args.category[0])
+        noteset.save()
 
-    main()
+    pid = is_running()
+    if pid: # send signal to the existing process accordingly
+        os.kill(pid, (args.kill and signal.SIGKILL) or (args.refresh and    \
+            signal.SIGUSR2) or signal.SIGUSR1)
+    elif not (args.kill or args.refresh):
+        if args.no_daemon: main()   # run
```

Comment:
My preference would be keeping the no-daemon behaviour by default. That is how most applications work.